### PR TITLE
docs: Cleanup README grammar, punctuation, and capitalization

### PR DESCRIPTION
**Pull request by dfsnow**
# Description

This PR cleans up some of the grammar, punctuation, and capitalization in the README. It also fixes some misspecified links (that I created! Oops), and tries to clarify some language. I've added some PR comments where I feel the README language is still unclear.

## Checklist

- [x] You followed [contributing section](https://github.com/dbt-athena/dbt-athena#contributing)
- [x] You kept your Pull Request small and focused on a single feature or bug fix.
- [ ] You added unit testing when necessary
- [ ] You added functional testing when necessary

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -826,8 +826,8 @@ def model(dbt, spark_session):
 
 - Python models cannot
  [reference Athena SQL views](https://docs.aws.amazon.com/athena/latest/ug/notebooks-spark.html).
-- Third-party Python libraries can be used, but they must be [included in the pre-installed list]([pre-installed list])
- or [imported manually]([imported manually]).
+- Third-party Python libraries can be used, but they must be [included in the pre-installed list][pre-installed list]
+ or [imported manually][imported manually].
```
Comment (dfsnow) [inline README.md:830]: I misspecified these links in #643. My mistake!

File: README.md [hunk 2]
```diff
@@ -704,8 +704,8 @@ The adapter supports python models using [`spark`](https://docs.aws.amazon.com/a
 - `spark_cross_account_catalog` (`default=false`)
   - In Spark, you can query the external account catalog and for that the consumer account has to be configured to
    access the producer catalog.
```
Comment (dfsnow) [inline README.md:706]: I find this sentence extremely unclear. Can anyone help me try to parse/clarify what it means so I can rewrite it?

Comment (Jrmyy) [reply README.md:707]: I would say something like this:

> When using Spark Athena workgroup, queries can only be run against catalogs located on the same AWS account. But sometimes you want to make queries to an other catalog located on an external AWS account. This requires an additional parameter on the spark properties setup. Set it to true when you want to query external catalogs.  If this flag is set to true, you can use the syntax `external_catalog_id/database.table` to access the external table on the external catalog (ex: 999999999999/mydatabase.cloudfront_logs` where 999999999999 is the external catalog ID)

And I will remove the next line

File: README.md [hunk 3]
```diff
@@ -841,13 +842,13 @@ def model(dbt, spark_session):
...
+The adapter partly supports contract definitions:
 
-- Concerning the `data_type`, it is supported but needs to be adjusted for complex types. They must be specified
+- `data_type` is supported but needs to be adjusted for complex types. Types must be specified
   entirely (for instance `array<int>`) even though they won't be checked. Indeed, as dbt recommends, we only compare
   the broader type (array, map, int, varchar). The complete definition is used in order to check that the data types
-  defined in athena are ok (pre-flight check).
-- the adapter does not support the constraints since no constraints don't exist in Athena.
+  defined in Athena are ok (pre-flight check).
+- The adapter does not support the constraints since no constraints don't exist in Athena.
```
Comment (Jrmyy) [inline README.md:851]:
```suggestion
- The adapter does not support the constraints since there is no constraint concept in Athena.
```

**Review by Jrmyy — COMMENTED** [review]
Thanks for all the cleaning :)

File: README.md (hunk 2 above)
Comment (dfsnow) [reply README.md]: Fixed in e5ac7db!

**Review by Jrmyy — APPROVED** [review]

**Pull request merged**